src/OmniSharp.Roslyn.CSharp/Services/Signatures/SignatureHelpService.cs

```diff
@@ -94,19 +113,19 @@ public SignatureHelpService(OmniSharpWorkspace workspace)
                 if (node is InvocationExpressionSyntax invocation && invocation.ArgumentList.Span.Contains(position))
                 {
                     var semanticModel = await document.GetSemanticModelAsync();
-                    return new InvocationContext(semanticModel, position, invocation.Expression, invocation.ArgumentList);
+                    return new InvocationContext(semanticModel, position, invocation.Expression, invocation.ArgumentList, invocation.IsInStaticContext());
```

Comment:
Can InvocationContext take care of calling IsInStaticContext()? You've making the same call in all the initializations.

Reply:
In that case we will have to pass the invocation object to InvocationContext and since there are three different types of such objects we will have to add another constructor in InvocationContext. Which is better of the two ?

Reply:
Ok, I'll leave it up to you.